faiss/IVFlib.cpp

```diff
@@ -60,23 +60,19 @@ void check_compatible_for_merge (const Index * index0,
 
 const IndexIVF * try_extract_index_ivf (const Index * index)
 {
+    if(index == nullptr) return nullptr;
     if (auto *pt =
         dynamic_cast<const IndexPreTransform *>(index)) {
         index = pt->index;
-    }
```

Comment:
This is on purpose so that a nested indexpretransform and an idmap can be extracted correctly.